package/network/services/dropbear/Makefile

```diff
@@ -89,45 +92,56 @@ CONFIGURE_ARGS += \
 TARGET_CFLAGS += -DARGTYPE=3 -ffunction-sections -fdata-sections -flto
 TARGET_LDFLAGS += -Wl,--gc-sections -flto=jobserver
 
+# remove when pull #786 will be upstreamed
+# ref: https://github.com/openwrt/openwrt/pull/786
+ESED:=$(STAGING_DIR_HOST)/bin/sed -E -i -e
+
 define Build/Configure
 	$(Build/Configure/Default)
 
-	$(SED) 's,^#define DEFAULT_PATH .*$$$$,#define DEFAULT_PATH "$(TARGET_INIT_PATH)",g' \
-		$(PKG_BUILD_DIR)/options.h
+	# disable legacy/unsafe methods and unused functionality
```

Comment:
Disabling DROPBEAR_SHA1_96_HMAC causes backwards compatibility issues with slightly older devices.
I tested this with another product type we have here, which is running a 2-year old dropbear version.
Making an ssh connection results in No matching algo mac c->s error.

In my opinion, this one should be enabled.

One could argue that it's required for safety, but:

- A lot of users don't have full (firmware) control of every device they use or remotely access
- It's a hard compat. break change without any prior notice

Reply:
@xback, thanks for report, i'll fix this.

Reply:
@rockdrilla DROPBEAR_SHA1_96_HMAC is disabled since March 2011 (3c801b3dc0359aa8b2b8876503f03e4bdf89c649). Please don't change that.

@xback maybe you confound this with DROPBEAR_SHA1_HMAC?

Reply:
@rockdrilla You shouldn't disable DROPBEAR_SHA1_HMAC, but disable DROPBEAR_MD5_HMAC instead. This is what it is done at the moment.

Reply:
@rockdrilla And maybe you shouldn't modify the file default_options.h at all, but instead add an localoptions.h like described in default_options.h.

Reply:
DROPBEAR_MD5_HMAC is disabled by default in sysoptions.h. So this change is obsolete.